src/main/java/com/yahoo/bullet/bql/extractor/ComputationExtractor.java

```diff
@@ -5,184 +5,100 @@
-import com.yahoo.bullet.bql.tree.Expression;
-import com.yahoo.bullet.bql.tree.Identifier;
-import com.yahoo.bullet.bql.tree.LongLiteral;
-import com.yahoo.bullet.bql.tree.Node;
-import com.yahoo.bullet.bql.tree.NullLiteral;
-import com.yahoo.bullet.bql.tree.ParensExpression;
-import com.yahoo.bullet.bql.tree.StringLiteral;
-import com.yahoo.bullet.parsing.BinaryExpression;
+import com.yahoo.bullet.bql.tree.ExpressionNode;
+import com.yahoo.bullet.bql.tree.SelectItemNode;
+import com.yahoo.bullet.bql.tree.SortItemNode;
 import com.yahoo.bullet.parsing.Computation;
-import com.yahoo.bullet.parsing.Expression.Operation;
-import com.yahoo.bullet.parsing.LeafExpression;
-import com.yahoo.bullet.parsing.PostAggregation;
-import com.yahoo.bullet.parsing.Value;
-import com.yahoo.bullet.typesystem.Type;
+import com.yahoo.bullet.parsing.Projection;
 
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
-
-import static java.util.Objects.requireNonNull;
+import java.util.stream.Stream;
 
 public class ComputationExtractor {
-    private List<Expression> computations;
-    private Map<Node, Identifier> aliases;
+    private ProcessedQuery processedQuery;
+
+    public Computation extractComputation(ProcessedQuery processedQuery) {
+        this.processedQuery = processedQuery;
+        switch (processedQuery.getQueryType()) {
+            case SELECT:
+                // No computations since everything is handled in the initial projection.
+                return null;
+            case SELECT_ALL:
+                return extractAll();
+            case SELECT_DISTINCT:
+                return extractDistinct();
+            case GROUP:
+            case COUNT_DISTINCT:
+            case DISTRIBUTION:
+            case TOP_K:
+                return extractAggregate();
+            case SPECIAL_K:
+                return extractSpecialK();
+        }
+        throw new ParsingException("Unreachable");
+    }
 
     /**
-     * Constructor that requires a set of selectFields and map of aliases.
-     *
-     * @param computations The non-null Set of computations.
-     * @param aliases      The non-null Map of aliases.
+     * Mirrors the projection logic. If the record is passed through, there should be no computations.
      */
-    public ComputationExtractor(List<Expression> computations, Map<Node, Identifier> aliases) {
-        requireNonNull(computations);
-        requireNonNull(aliases);
-        this.computations = computations;
-        this.aliases = aliases;
+    private Computation extractAll() {
+        List<Projection.Field> fields =
+                Stream.concat(processedQuery.getSelectNodes().stream().map(SelectItemNode::getExpression),
+                              processedQuery.getOrderByNodes().stream().map(SortItemNode::getExpression))
+                        .filter(expression -> processedQuery.isNotSimpleFieldExpression(expression) || processedQuery.hasAlias(expression))
```

Comment:
Uh, in this case, it's not what we can't do, but what we need to or don't need to do. This is the case where the query is SELECT *.

If it's just a simple field expression, there's no need to compute anything. It's already in the record. 
However, if it has an alias, e.g. SELECT *, abc AS def ... , then we need to put "def" in the record. 

It'd be nice to just put it in the initial projection, to be honest, and we can. It's just that the current logic (that I added in bullet-core) for copying the bullet record is to have a Projection with null fields (whereas a null Projection means to just pass the bullet record as is). Instead of doing the null fields, it might be better to have a boolean flag "copy" or something.